kkm/kkm_kontext.c

```diff
@@ -278,6 +279,7 @@ int kkm_kontext_switch_kernel(struct kkm_kontext *kkm_kontext)
 				     .kvaddr;
 			if (pa->reason == FAULT_SYSCALL) {
 				ga->regs.rsp += sizeof(struct kkm_hc_args);
+				ga->regs.rsp += KKM_ABI_REDZONE;
```

Comment:
Is the compiler able to optimize this down to a single addition to regs.rsp?

If not, could you change this to:

ga->regs.rsp += sizeof(struct kkm_hc_args) + KKM_ABI_REDZONE;

Same question where we subtract from the stack in kkm_process_syscall().